R/timeseriessummary-data.R

```diff
@@ -524,6 +525,178 @@ parseTSSGapTolerances <- function(reportData, timezone){
+                                   "FillGaps" = formatCorrectionsParamFillGaps(ResamplePeriod, GapLimit),
+                                   "Deviation" = formatCorrectionsParamDeviation(DeviationValue, DeviationType, WindowSizeInMinutes),
+                                   type)) %>% 
+    ungroup()
+  
+  
+  return(corrections_formatted)
+}
+
+#' formats the offset correction parameters
+#' @description formats the offset correction parameters
+#' @param offset the offset value for the parameter offset
+#' @return formatted string of offset parameters for report display
+formatCorrectionsParamOffset <- function(offset) {
+  formattedParameters <- ""
+  if (!isEmptyOrBlank(offset)) {
+    formattedParameters <- paste0("Offset ", offset)
+  }
+  return(formattedParameters)
+}
+
+#' formats the drift correction parameters
+#' @description formats the drift correction parameters
+#' @param driftPoints the driftPoints values in a list including
+#' date/times and the difference values in feet
+#' @param timezone the timezone for the drift parameters
+#' @return formatted string of drift parameters for report display
+formatCorrectionsParamDrift <- function(driftPoints, timezone) {
+  formattedParameters <- ""
+  if (!isEmptyOrBlank(driftPoints) && !isEmptyOrBlank(timezone)) {
+    driftPoints <- as.data.frame(driftPoints)
+    if (any(names(driftPoints) %in% c("Value","Offset"))) {
+        for (i in 1:nrow(driftPoints)) {
+          formattedParameters <- paste0(formattedParameters, "Drift correction of (date/time, diff): (", flexibleTimeParse(driftPoints[['Time']][[i]], timezone, FALSE), ", ", driftPoints[['Offset']][[i]], "ft. ")
+        }
+    }
+  }
+  return(formattedParameters)
+}
+
+#' formats the single point correction parameters
+#' @description formats the single point correction parameters
+#' @param value the value for the parameter single point
+#' @return formatted string of single point parameter for report display
+formatCorrectionsParamSinglePoint <- function(value) {
+  formattedParameters <- ""
+  if (!isEmptyOrBlank(value)) {
+    formattedParameters <- paste0("SinglePoint ", value)
+  }
+  return(formattedParameters)
+}
+
+#' formats the USGSMultiPoint correction parameters
+#' @description formats the USGSMultiPoint correction parameters
+#' @param startShiftPoints startShift points values and offset as a list
+#' @param endShiftPoints endShift points values and offset as a list
+#' @param usgsType the USGSMultiPoint type
+#' @return formatted string of USGSMultiPoint parameters for report display
+formatCorrectionsParamUSGSMultiPoint <- function(startShiftPoints, endShiftPoints, usgsType) {
+  formattedParameters <- "USGSMultiPoint "
+  if (!isEmptyOrBlank(startShiftPoints) && !isEmptyOrBlank(endShiftPoints) && !isEmptyOrBlank(usgsType)) {
+    startShiftPoints <- as.data.frame(startShiftPoints)
+    endShiftPoints <- as.data.frame(endShiftPoints)
+    if (any(names(startShiftPoints) %in% c("Value","Offset"))) {
```

Comment:
you use `any` here for start and end points, but do you need it to include both? Both are used in the paste functions. Maybe you need `all` instead? Then you just switch the order: `all(c("Value","Offset") %in% names(startShiftPoints))`